tests/integration/kubernetes/runtimeclass_workloads/pod-hostname.yaml

```
@@ -0,0 +1,18 @@
+#
+# Copyright (c) 2024 Ant Group.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-pod-hostname
+spec:
+  terminationGracePeriodSeconds: 0
+  runtimeClassName: kata
+  containers:
+  - image: quay.io/prometheus/busybox:latest 
+    name: pod-hostname-c01
+    command: ["sh"]
+    args: ["-c", "hostname && sleep infinity"]
```

nit: I don't think we need to sleep here, we should still be able to get the logs.
```suggestion
    args: ["-c", "hostname"]
```

---

```shell
### If it's with `&& sleep infinity`,get result as expected.
root@XXX:~/debug-kata/examples# kubectl apply -f pod-hostname.yaml 
pod/pod01-hostname created
root@XXX:~/debug-kata/examples# kubectl wait --for=condition=Ready --timeout=10s pod pod01-hostname
pod/pod01-hostname condition met
root@XXX:~/debug-kata/examples# kubectl get po
NAME             READY   STATUS    RESTARTS   AGE
pod01-hostname   1/1     Running   0          16s

### If it is without `&& sleep infinity` 
root@XXX:~/debug-kata/examples# kubectl apply -f pod-hostname.yaml 
pod/pod01-hostname created
root@XXX:~/debug-kata/examples# kubectl get po
NAME             READY   STATUS              RESTARTS   AGE
pod01-hostname   0/1     ContainerCreating   0          3s
root@XXX:~/debug-kata/examples# kubectl get po
NAME             READY   STATUS      RESTARTS     AGE
pod01-hostname   0/1     Completed   1 (2s ago)   6s
root@XXX:~/debug-kata/examples# kubectl wait --for=condition=Ready --timeout=10s pod pod01-hostname
error: timed out waiting for the condition on pods/pod01-hostname
```

---

@Apokleos  `kubectl wait` isn't needed since you don't do the `kubectl exec` thing anymore.